manifests/init.pp

```diff
@@ -73,7 +74,7 @@
 class erlang (
   String  $package_name   = 'erlang',
   String  $package_ensure = 'installed',
-  Optional[Variant[Numeric, String]] $package_apt_pin = undef,
+  Optional[Variant[Numeric, String]] $package_apt_pin = 200,
```

Comment:
If it's still not recommended to have a default value for optional parameters, open to suggestions on the best way to proceed here...

Reply:
Then I'd change the data type. I also wonder why it's a `Numeric`. Isn't a pin always an integer? https://manpages.debian.org/buster/apt/apt_preferences.5.en.html#How_APT_Interprets_Priorities suggests it is.
```suggestion
  Variant[Integer, String] $package_apt_pin = 200,
```

Reply:
That was my thought as well (changing data type), but are there use cases where someone would want / need to override it to being unset? I can just adjust the docs and make it non-optional as long as everyone’s good with that.

Reply:
It can still be unset by setting it to `undef`.

Reply:
The data types would also need to be changed in `repo/apt/bintray.pp` and `repo/apt/erlang_solutions.pp`.

Reply:
@kenyon at one point, you could not set a default value for an optional parameter. I'm not sure if that's still the case, though

Reply:
@wyardley my comment was under the assumption you implement @ekohl's suggestion, so it wouldn't be optional.